Library/Homebrew/cmd/sh.rb

```diff
@@ -13,7 +13,7 @@ def sh
     ENV.activate_extensions!
 
     if superenv?
-      ENV.x11 = MacOS::X11.installed?
+      ENV.x11_if_installed
```

Comment:
I'm not sure. I never used `brew sh` and I'm not entirely sure what that line is supposed to do (in more broader terms, I understood your explanation from above).